x-pack/legacy/plugins/ml/public/application/explorer/actions/load_explorer_data.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import memoizeOne from 'memoize-one';
+import { isEqual } from 'lodash';
+
+import { forkJoin, of } from 'rxjs';
+import { mergeMap, tap } from 'rxjs/operators';
+
+import { explorerChartsContainerServiceFactory } from '../explorer_charts/explorer_charts_container_service';
+import { VIEW_BY_JOB_LABEL } from '../explorer_constants';
+import { explorerService } from '../explorer_dashboard_service';
+import {
+  getDateFormatTz,
+  getSelectionInfluencers,
+  getSelectionTimeRange,
+  loadAnnotationsTableData,
+  loadAnomaliesTableData,
+  loadDataForCharts,
+  loadFilteredTopInfluencers,
+  loadOverallData,
+  loadTopInfluencers,
+  loadViewBySwimlane,
+  loadViewByTopFieldValuesForSelectedTime,
+} from '../explorer_utils';
+import { ExplorerState } from '../reducers';
+
+const memoizeIsEqual = (newArgs: any[], lastArgs: any[]) => isEqual(newArgs, lastArgs);
+
+// Memoize the data fetching methods
+// TODO: We need to track an attribute that allows refetching when the date picker
+// triggers a refresh, otherwise we'll get back the stale data. Note this was also
+// an issue with the previous version and the custom caching done within the component.
+const memoizedLoadAnnotationsTableData = memoizeOne(loadAnnotationsTableData, memoizeIsEqual);
+const memoizedLoadDataForCharts = memoizeOne(loadDataForCharts, memoizeIsEqual);
+const memoizedLoadFilteredTopInfluencers = memoizeOne(loadFilteredTopInfluencers, memoizeIsEqual);
+const memoizedLoadOverallData = memoizeOne(loadOverallData, memoizeIsEqual);
+const memoizedLoadTopInfluencers = memoizeOne(loadTopInfluencers, memoizeIsEqual);
+const memoizedLoadViewBySwimlane = memoizeOne(loadViewBySwimlane, memoizeIsEqual);
+const memoizedLoadAnomaliesTableData = memoizeOne(loadAnomaliesTableData, memoizeIsEqual);
+
+const dateFormatTz = getDateFormatTz();
+
+/**
+ * Fetches the data necessary for the Anomaly Explorer using observables.
+ *
+ * @param state ExplorerState
+ *
+ * @return Partial<ExplorerState>
+ */
+export function loadExplorerData(state: ExplorerState) {
```

Comment:
nit: the return type is better to define in the function itself. it'll also help to valiate your function 
```suggestion
export function loadExplorerData(state: ExplorerState):  Partial<ExplorerState> {
```